src/game/AI/ScriptDevAI/scripts/outland/netherstorm.cpp

```diff
@@ -181,7 +181,7 @@ struct npc_manaforge_spawnAI : public ScriptedAI
         Creature *manaforge;
 
         if (m_manaforgeGuid)
-            if (manaforge = m_creature->GetMap()->GetCreature(m_manaforgeGuid))
+            if ((manaforge = m_creature->GetMap()->GetCreature(m_manaforgeGuid)))
```

Comment:
I understand that you tried to make compiler happy there.
But this way to test the result of the GetCreature() result is not good to me and its the main issue here.
I would suggest to broke that in 2 instructions

```cpp
if (m_manaforgeGuid)
{
     manaforge = m_creature->GetMap()->GetCreature(m_manaforgeGuid);
     if (manaforge)
     {
          .....
     }
}

```

Reply:
I had it that way before that, but I saw that several other similar comparisons were doing it in this style, so I adapted it there, too